unittests/optionaltype_pointer/testpointerapi.cpp

```diff
@@ -39,4 +40,20 @@ void TestPointerApi::test()
     QCOMPARE(*resp.out(), newVal);
 }
 
+void TestPointerApi::testOptionalArray()
+{
+    TNS__TestOperationResponse1 resp;
+    QCOMPARE(resp.outOptionalArray(), (QList<qint64> *)0);
+    const QList<qint64> valIn(QList<qint64>() << 1 << 2 << 3);
+    resp.setOutOptionalArray(valIn);
+
+    TNS__TestOperationResponse1 resp2;
+    resp2.deserialize(resp.serialize("array"));
+
+    QVERIFY(resp2.outOptionalArray());
+    if (resp2.outOptionalArray()) {
```

Comment:
This if() isn't needed, since QVERIFY returns on error. So if we're here, the condition in this if() can't ever be false.